drivers/gpio/Kconfig.ad5592

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2023 Grinn
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_AD5592
+	bool "AD5592 GPIO driver"
+	default y
+	depends on DT_HAS_ADI_AD5592_GPIO_ENABLED
+	select MFD
+	help
+	  Enable the AD5592 GPIO driver.
+
+config GPIO_AD5592_INIT_PRIORITY
```

Comment:
Hey how about using `CONFIG_MFD_INIT_PRIORITY` for the subdevices as well? Then the sublevel will ensure that the parent is initialized first automatically. We haven't done that for other MFD devices, but I'm thinking we should.

Reply:
not totally convinced about this (yet) to be honest but worth having a thought about

Reply:
Let's find out ;)

Reply:
@fabiobaltieri it seems you were right. I'm gonna remove the extra `*_INIT_PRIORITY` options.

Reply:
cool!

Reply:
guess an more conservative alternative could have been to default them to `MFD_INIT_PRIORITY` but if we can just get rid of them it's even better, maybe we can drop other MFD child symbols too

Reply:
Done.